### PR TITLE
FIR: Do not load `inline` flag when deserializing properties

### PR DESCRIPTION
**Pull request by punzki**
...there is no `inline` flag for serialized properties.

In order to test this, I added the changes to FirRenderer to make sure the flag is not loaded. However, this revealed that the `inline` status was propagated upward to the `FirProperty` during raw FIR building, causing test failures.

I removed the upward propagation for `inline`. I also removed it for `external` because it is incorrect: `external` on properties (used in JS) should be separate from `external` on accessors (used in JNI interop for JVM).  The `external` flags are also serialized separately for properties and accessors.

### CODE REVIEW

File: compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt [hunk 1]
```diff
@@ -52,8 +52,19 @@ internal fun FirMemberDeclaration.isEffectivelyExternal(
...
+    if (this is FirPropertyAccessor) {
+        // Check containing property
+        val property = context.containingDeclarations.last() as FirProperty
+        return property.isEffectivelyExternal(containingClass, context)
+    }
+
+    if (this is FirProperty) {
+        // Property is effectively external if all accessors are external
+        if (getter?.isExternal == true &&
+            (!isVar || setter?.isExternal == true)) {
```
Comment (jsjeon) [inline compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt:64]: Nit: not formatted.

**Review by jsjeon — APPROVED** [review]

**Comment by punzki** [comment]
@semoro Can you please take a look?

File: compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt [hunk 2]
```diff
@@ -49,12 +49,6 @@ inline val FirRegularClass.isAbstract get() = status.modality == Modality.ABSTRA
 inline val FirRegularClass.canHaveAbstractDeclaration: Boolean
     get() = isAbstract || isSealed || isEnumClass
 
-inline val FirRegularClass.isInner get() = status.isInner
```
Comment (semoro) [inline compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt:52]: Why those functions should be allowed on FirMemberDeclaration? 
It is applicable only to classes

File: compiler/fir/raw-fir/psi2fir/src/org/jetbrains/kotlin/fir/builder/RawFirBuilder.kt [hunk 3]
```diff
@@ -1263,17 +1263,13 @@ open class RawFirBuilder(
...
-                        // Upward propagation of `inline` and `external` modifiers (from accessors to property)
-                        // Note that, depending on `var` or `val`, checking setter's modifiers should be careful: for `val`, setter doesn't
-                        // exist (null); for `var`, the retrieval of the specific modifier is supposed to be `true`
                         status = FirDeclarationStatusImpl(visibility, modality).apply {
                             isExpect = hasExpectModifier() || containingClassOrObject?.hasExpectModifier() == true
                             isActual = hasActualModifier()
                             isOverride = hasModifier(OVERRIDE_KEYWORD)
                             isConst = hasModifier(CONST_KEYWORD)
                             isLateInit = hasModifier(LATEINIT_KEYWORD)
-                            isInline = hasModifier(INLINE_KEYWORD) || (getter!!.isInline && setter?.isInline != false)
```
Comment (semoro) [inline compiler/fir/raw-fir/psi2fir/src/org/jetbrains/kotlin/fir/builder/RawFirBuilder.kt:1275]: inline flag isn't set at all now?

File: compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/converter/DeclarationsConverter.kt [hunk 4]
```diff
@@ -1033,17 +1033,13 @@ class DeclarationsConverter(
...
-                    // Upward propagation of `inline` and `external` modifiers (from accessors to property)
-                    // Note that, depending on `var` or `val`, checking setter's modifiers should be careful: for `val`, setter doesn't
-                    // exist (null); for `var`, the retrieval of the specific modifier is supposed to be `true`
                     status = FirDeclarationStatusImpl(propertyVisibility, modifiers.getModality()).apply {
                         isExpect = modifiers.hasExpect() || classWrapper?.hasExpect() == true
                         isActual = modifiers.hasActual()
                         isOverride = modifiers.hasOverride()
                         isConst = modifiers.isConst()
                         isLateInit = modifiers.hasLateinit()
-                        isInline = modifiers.hasInline() || (getter!!.isInline && setter?.isInline != false)
```
Comment (semoro) [inline compiler/fir/raw-fir/light-tree2fir/src/org/jetbrains/kotlin/fir/lightTree/converter/DeclarationsConverter.kt:1045]: inline flag isn't set at all now?

File: compiler/fir/fir-deserialization/src/org/jetbrains/kotlin/fir/deserialization/FirMemberDeserializer.kt [hunk 5]
```diff
@@ -305,7 +305,6 @@ class FirMemberDeserializer(private val c: FirDeserializationContext) {
                 isOverride = false
                 isConst = Flags.IS_CONST.get(flags)
                 isLateInit = Flags.IS_LATEINIT.get(flags)
-                isInline = Flags.IS_INLINE.get(flags)
```
Comment (semoro) [inline compiler/fir/fir-deserialization/src/org/jetbrains/kotlin/fir/deserialization/FirMemberDeserializer.kt:308]: Why inline flag shouldn't be loaded?

**Review by semoro — CHANGES_REQUESTED** [review]

File: compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt (hunk 2 above)
Comment (punzki) [reply compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt:52]: Sorry for the delayed reply, I was out of office and I missed the email reminder.

In order to test these changes, I made changes to `FirRenderer` to make sure the `inline` flag is not loaded. I decided to render all the status flags as well for all declarations, even if they are not applicable, just to make sure no incorrect flags are loaded for any of them.

Comment (semoro) [reply compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt:51]: In such a case, I recommend not to change those functions but instead check for status explicitly in the renderer
As those functions are part of a wide-used API

Comment (punzki) [reply compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt:51]: Good idea! Done. I'm also rendering `fun` for completeness.

In my defense, there are other util functions there that aren't specific to the appropriate declaration type (e.g., `isTailRec` should only be for `FirSimpleFunction`, `isLateInit` for `FirProperty`) but I won't touch those right now.

Comment (punzki) [reply compiler/fir/tree/src/org/jetbrains/kotlin/fir/declarations/FirDeclarationUtil.kt:51]: @semoro As discussed in our sync, I am not rendering `fun` for Java interfaces. Updated the PR.

**Comment by punzki** [comment]
@semoro Gentle ping. Let me know if there are any other changes you are looking for!

**Comment by semoro** [comment]
Sent to auto-merge

**Comment by jsjeon** [comment]
Just out of curiosity, can auto-merge automatically merge conflicting files too? Otherwise, the PR/commits need to be updated before being sent to auto-merge?

**Comment by punzki** [comment]
@semoro I rebased and resolved conflicts. It might have been preventing the auto-merge. Do we need to do anything else to send it back to auto-merge?

**Comment by punzki** [comment]
@semoro What is the status of this PR merge?

**Comment by punzki** [comment]
Closing as it looks like it's already been merged: https://github.com/JetBrains/kotlin/commits/578fcf2ebf11886bbfe3a27a3ab4da150e39fdb9

**Pull request closed without merging**